oci/archive/oci_dest.go

```diff
@@ -0,0 +1,188 @@
+package archive
+
+import (
+	"io"
+	"os"
+
+	"github.com/containers/image/types"
+	"github.com/containers/storage/pkg/archive"
+	"github.com/pkg/errors"
+)
+
+type ociArchiveImageDestination struct {
+	ref          ociArchiveReference
+	unpackedDest types.ImageDestination
+	tempDir      string
+}
+
+// newImageDestination returns an ImageDestination for writing to an existing directory.
+func newImageDestination(ref ociArchiveReference, ctx *types.SystemContext) (types.ImageDestination, error) {
+	tempDirOciRef, err := createOciRef(ref.image)
```

Comment:
(This creates an one-time named structure, unpacks it in the same function, and then throws it away. Perhaps `createOciRef` could just return a tuple of three values (reference, path, error) instead of bothering to define and return a structure.

OTOH structures with named members are more explicit than tuples with positional results, perhaps it is worth it. Or if `deleteTempDir` became a method on the structure.)